cmd/minishift/cmd/image/list.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"fmt"
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/profile"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dockerDaemonImages bool
+
+	imageCacheListCmd = &cobra.Command{
+		Use:   "list ",
+		Short: "Displays the locally cached images.",
+		Long:  "Displays the locally cached images.",
+		Run:   listImages,
+	}
+)
+
+func listImages(cmd *cobra.Command, args []string) {
+	api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+	defer api.Close()
+
+	if dockerDaemonImages {
+		listDockerDaemonImages(api)
+	} else {
+		listCachedImages()
+	}
+}
+
+func listCachedImages() {
+	cacheDir := constants.MakeMiniPath(CacheDir...)
+	cachedImages := getCachedImages(cacheDir)
+	if len(cachedImages) == 0 {
+		fmt.Println(fmt.Sprintf("There are no images cached for profile '%s' in '%s'", profile.GetActiveProfile(), cacheDir))
```

Comment:
We need to use constants.ProfileName instead of profile.GetActiveProfile() . Or we can fix https://github.com/minishift/minishift/issues/1769 first and then you can update your PR.

Reply:
I am actually confused about the naming that has been suggested in #1769. I am not sure introducing the function 'GetCurrentProfile' is clear to describe its intentions...

Reply:
> We need to use constants.ProfileName instead of profile.GetActiveProfile() 

Ok, I changed it. For what it's worth, I agree with @gbraad that the naming is off here. I was actually thinking about this when I wrote the code.